contract/src/main/java/io/neow3j/contract/NFToken.java

```diff
@@ -0,0 +1,261 @@
+     * @param amount the fraction to be transferred.
+     * @param tokenId the script hash of the tokenId.
+     * @return a transaction builder.
+     * @throws IOException if there was a problem fetching information from the Neo node.
+     */
+    public TransactionBuilder transferFraction(Wallet wallet, ScriptHash from, ScriptHash to,
+            BigDecimal amount, ScriptHash tokenId) throws IOException {
+        List<ScriptHash> owners = ownerOf(tokenId);
+
+        if (owners.stream().noneMatch(from::equals)) {
+            throw new IllegalStateException("The account " + from.toAddress() + " is not an owner " +
+                    "of the token with script hash " + tokenId.toString() + ".");
+        }
+
+        if (!wallet.holdsAccount(from)) {
+            throw new IllegalArgumentException("The provided wallet does not contain the provided from " +
+                    "account.");
+        }
+
+        BigInteger fractions = getAmountAsBigInteger(amount);
+        System.out.println(fractions);
+
+        return invokeFunction(TRANSFER,
+                ContractParameter.hash160(from),
+                ContractParameter.hash160(to),
+                ContractParameter.integer(fractions),
+                ContractParameter.hash160(tokenId))
+                .wallet(wallet)
+                .signers(Signer.calledByEntry(from));
+    }
+
+    /**
+     * Gets the owner(s) of the token with script hash {@code tokenId}.
+     *
+     * @param tokenId the script hash of the tokenId.
+     * @return a transaction builder
+     */
+    // According to the not yet final NEP-11 proposal, this method returns an enumerator that contains all
+    //  the co-owners that own the specified token. However, there no stack item type enumerator is
+    //  implemented yet. Hence, this method expects an array.
+    // TODO: 15.10.20 Michael: Adapt this method as soon as an implementation of the NEP-11 proposal exists.
+    public List<ScriptHash> ownerOf(ScriptHash tokenId) throws IOException {
+        return callFunctionReturningListOfScriptHashes(OWNER_OF,
+                Arrays.asList(ContractParameter.hash160(tokenId)));
+    }
+
+    private List<ScriptHash> callFunctionReturningListOfScriptHashes(String function,
+            List<ContractParameter> params) throws IOException {
+
+        StackItem arrayItem = callInvokeFunction(function, params).getInvocationResult().getStack().get(0);
+        if (!arrayItem.getType().equals(StackItemType.ARRAY)) {
+            throw new UnexpectedReturnTypeException(arrayItem.getType(), StackItemType.ARRAY);
+        }
+        List<ScriptHash> scriptHashes = new ArrayList<>();
+        for (StackItem item : arrayItem.asArray().getValue()) {
+            scriptHashes.add(extractScriptHash(item));
+        }
+        return scriptHashes;
+    }
+
+    private ScriptHash extractScriptHash(StackItem item) {
+        if (!item.getType().equals(StackItemType.BYTE_STRING)) {
```

Comment:
The `item` would ever be null? If not, it's fine. Otherwise, add a check for it.

Reply:
Moreover, check [this](https://codecov.io/gh/neow3j/neow3j/src/772ef549226ff79a642158b8833f96a69b9512d3/contract/src/main/java/io/neow3j/contract/NFToken.java).

This is the report of codecov. It seems we're not testing the cases when the exception is thrown (some occurrences throughout the code). My question is: testing this kind of thing is valuable? Or just cosmetics?

If you think it adds value, it's worth adding tests for the exception cases.

Reply:
BTW: these comments are not specific this line, but it's kind of generic for our repo. 👍

Reply:
> The `item` would ever be null? If not, it's fine. Otherwise, add a check for it.

I added a check for it.

> Moreover, check [this](https://codecov.io/gh/neow3j/neow3j/src/772ef549226ff79a642158b8833f96a69b9512d3/contract/src/main/java/io/neow3j/contract/NFToken.java).
> 
> This is the report of codecov. It seems we're not testing the cases when the exception is thrown (some occurrences throughout the code). My question is: testing this kind of thing is valuable? Or just cosmetics?
> 
> If you think it adds value, it's worth adding tests for the exception cases.

I will have a look at this. 👍

Reply:
As we just discussed, let's leave like that for the moment. 👍 💯